File: 백준/Silver/11727. 2×n 타일링 2/2×n 타일링 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[] sequence = new int[1000];
        sequence[0] = 1;
        sequence[1] = 3;
        for (int i = 0; i < N; i++) {
            if (i < 2) continue;
            sequence[i] = (sequence[i - 1] + 2 * sequence[i - 2]) % 10007;
        }
        System.out.println(sequence[N - 1]);
    }
}

File: 백준/Silver/11727. 2×n 타일링 2/2×n 타일링 2.cc
#include <iostream>
using namespace std;

int main() {
    int N;
    cin >> N;

    int sequence[1000];
    sequence[0] = 1;
    sequence[1] = 3;

    for (int i = 2; i < N; i++) {
        sequence[i] = (sequence[i - 1] + 2 * sequence[i - 2]) % 10007;
    }

    cout << sequence[N - 1] << endl;

    return 0;
}